org.eclipse.lsp4e/src/org/eclipse/lsp4e/LanguageServiceAccessor.java

```diff
@@ -279,110 +274,130 @@ private static boolean capabilitiesComply(LanguageServerWrapper wrapper,
 	 *
 	 * @param file
 	 * @param request
-	 * @return
+	 * @return the matching LS wrappers
 	 * @throws IOException
 	 * @noreference This method is currently internal and should only be referenced
 	 *              for testing
 	 */
 	@NonNull
-	public static Collection<LanguageServerWrapper> getLSWrappers(@NonNull IFile file,
-			@Nullable Predicate<ServerCapabilities> request) throws IOException {
-		LinkedHashSet<LanguageServerWrapper> res = new LinkedHashSet<>();
-		IProject project = file.getProject();
+	public static Collection<LanguageServerWrapper> getLSWrappers(@NonNull final IFile file,
+			@Nullable final Predicate<ServerCapabilities> request) throws IOException {
+		final var project = file.getProject();
 		if (project == null) {
-			return res;
+			return Collections.emptyList();
 		}
 
-		res.addAll(getMatchingStartedWrappers(file, request));
+		final var res = new LinkedHashSet<LanguageServerWrapper>(getMatchingStartedWrappers(file, request));
 
 		// look for running language servers via content-type
-		Queue<IContentType> contentTypes = new LinkedList<>();
-		Set<IContentType> addedContentTypes = new HashSet<>();
-		contentTypes.addAll(LSPEclipseUtils.getFileContentTypes(file));
-		addedContentTypes.addAll(contentTypes);
+		final var contentTypes = new ArrayDeque<IContentType>(LSPEclipseUtils.getFileContentTypes(file));
+		final var processedContentTypes = new HashSet<IContentType>();
```

Comment:
could we save LSPEclipseUtils.getFileContentTypes(file) into a local variable so that it can be used to initialize the capacity of processedContentTypes?